WebUI - Improved "dark" theme

facuxt opened the pull request:
TODO: tune rrdtool to display graphs in dark mode.
![image](https://user-images.githubusercontent.com/7550822/79134075-7e37df00-7d83-11ea-8b87-cbf52908d6ad.png)

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

Jellyfrog commented on app/Http/ViewComposers/MenuComposer.php:
```diff
@@ -62,7 +62,7 @@ public function compose(View $view)
         // end global Settings
 
         //TODO: should be handled via CSS Themes
-        $vars['navbar'] = in_array($site_style, ['mono', 'dark']) ? 'navbar-inverse' : '';
```
if this is no longer needed it should be removed from all templates and so that use it.

facuxt replied:
agree, i dont know where its used that variable... so i didnt removed it completely.

Jellyfrog replied:
Remove it and search for {{ $navbar}} to start with.

SourceDoctor replied:
if it's really not more used, for my intention TODO-Line should be removed also

facuxt replied:
Already fixed it, its needed by the "mono"theme... so i left that sentence, with "mono" only.

kkrumm1 commented:
Looks great! needs some tweaks here and there.  Also, "mono" is used as a theme. 

![Screen Shot 2020-04-16 at 7 42 42 AM](https://user-images.githubusercontent.com/27959419/79457539-fede2180-7fb5-11ea-9127-47612dec5eb0.png)

kkrumm1 commented:
When I apply your changes and use mono it breaks my navbar.
![Screen Shot 2020-04-16 at 7 46 11 AM](https://user-images.githubusercontent.com/27959419/79457795-6b592080-7fb6-11ea-90d2-98af9b22fa94.png)

kkrumm1 commented:
![Screen Shot 2020-04-16 at 7 47 10 AM](https://user-images.githubusercontent.com/27959419/79457864-8461d180-7fb6-11ea-8c3f-67b7473c601e.png)
Nav bar ^^ with mono theme. This needs to stay.

kkrumm1 commented:
@facuxt Anyword on this?

facuxt commented:
Im really sorry i cannot continue working on this, i have too much work at the moment, i will try to get this done by the weekend.

kkrumm1 commented:
No worries.

facuxt commented:
What do you think about this background color in the widgets. i think, the next step should be mess with the rddtool to make inverted-color graphs.
![image](https://user-images.githubusercontent.com/7550822/80016539-a02f1100-84a9-11ea-9834-c3d2ebf6f85e.png)

Does anyone knows how to make rddtool to make inverted color graphs...   any suggestion is valid here.

facuxt commented:
With the last commit, i fixed the mono theme navbar, and removed the red color from some of the icons, make the background of the gridster darker, and fixed an issue with some links in the tables.
![image](https://user-images.githubusercontent.com/7550822/80020095-dd49d200-84ae-11ea-87e1-2e00a780ca2c.png)
![image](https://user-images.githubusercontent.com/7550822/80020125-e9359400-84ae-11ea-95be-3613a47a348b.png)
![image](https://user-images.githubusercontent.com/7550822/80020187-ff435480-84ae-11ea-86bd-4531f134e6d2.png)

svenvg93 commented:
> With the last commit, i fixed the mono theme navbar, and removed the red color from some of the icons, make the background of the gridster darker, and fixed an issue with some links in the tables.
> ![image](https://user-images.githubusercontent.com/7550822/80020095-dd49d200-84ae-11ea-87e1-2e00a780ca2c.png)
> ![image](https://user-images.githubusercontent.com/7550822/80020125-e9359400-84ae-11ea-95be-3613a47a348b.png)
> ![image](https://user-images.githubusercontent.com/7550822/80020187-ff435480-84ae-11ea-86bd-4531f134e6d2.png)

Looks Awesome! 
Would be nice to have the RDD text in with, because now you cannot read them.

facuxt commented:
> > With the last commit, i fixed the mono theme navbar, and removed the red color from some of the icons, make the background of the gridster darker, and fixed an issue with some links in the tables.
> > ![image](https://user-images.githubusercontent.com/7550822/80020095-dd49d200-84ae-11ea-87e1-2e00a780ca2c.png)
> > ![image](https://user-images.githubusercontent.com/7550822/80020125-e9359400-84ae-11ea-95be-3613a47a348b.png)
> > ![image](https://user-images.githubusercontent.com/7550822/80020187-ff435480-84ae-11ea-86bd-4531f134e6d2.png)
> 
> Looks Awesome!
> Would be nice to have the RDD text in with, because now you cannot read them.
yes, but i cannot find a way to do that... Maybe some of the masters here knows a way 
@murrant

facuxt commented:
I found it... i made some changes in the color scheme in the graphs... now the look awesome!
![image](https://user-images.githubusercontent.com/7550822/80280829-17002000-86dd-11ea-9df5-a159c62b752e.png)
![image](https://user-images.githubusercontent.com/7550822/80280960-043a1b00-86de-11ea-8e9c-7fa507ddf5e2.png)

breeman1 commented:
This is lovely. Just a suggestion from my side; will it not be better to also make the color of the other icons the same as those in the navigation. They are almost invisible in their current black color, see below.
![Snap1](https://user-images.githubusercontent.com/16717388/80281373-e636df80-870a-11ea-8bc1-82445d8fb921.png)

breeman1 commented:
Perhaps even color of the font (white) in the sub menu can be made to be same as that of the navigation menu (gray) to improve visual coherence throughout the theme.

facuxt commented:
> Perhaps even color of the font (white) in the sub menu can be made to be same as that of the navigation menu (gray) to improve visual coherence throughout the theme.

You are right, i've made a commit with your suggested changes and some more..

- Icons and links now are grey
![image](https://user-images.githubusercontent.com/7550822/80282308-17051d80-86e7-11ea-8088-5e81a2b7f2c3.png)
- Modernized "tabs" style into pills, 
![image](https://user-images.githubusercontent.com/7550822/80282319-21271c00-86e7-11ea-8d4b-673e810a3698.png)
- button pagemenu-selected fixed
![image](https://user-images.githubusercontent.com/7550822/80282280-fd63d600-86e6-11ea-9b1a-53dcda698d88.png)

breeman1 commented:
Looks really cool, thanks.

breeman1 commented:
Just a quick question. How will this theme deal device symbols that are black and white i.e pfSense? I am just concerned they might be invisible. Below is how the pfSense symbol is currently displayed in our production install.
![image](https://user-images.githubusercontent.com/16717388/80358143-40a37d80-887c-11ea-8627-ab958e3229cd.png)

facuxt commented:
> Just a quick question. How will this theme deal device symbols that are black and white i.e pfSense? I am just concerned they might be invisible. Below is how the pfSense symbol is currently displayed in our production install.
> ![image](https://user-images.githubusercontent.com/16717388/80358143-40a37d80-887c-11ea-8627-ab958e3229cd.png)

i think i could add a css class to add a little white glow around the logos...  what is your opinion about that?

breeman1 commented:
Thanks, that could work I think. Would you mind posting a screenshot of what it could look like?

facuxt commented:
> Thanks, that could work I think. Would you mind posting a screenshot of what it could look like?

![image](https://user-images.githubusercontent.com/7550822/80415652-364ba880-88a9-11ea-950b-132ae3b3fda8.png)

like this without touching anything... see the part of the ubiquity logo that its black... i dont think its necessary to add a 1990 glow on it 😋

breeman1 commented:
I think it is because the eye/brain realizes the fact that the black is part of the gray and blue, making the need for the glow unnecessary in this case. Not sure about an all-black logo though, I think it will look strange without the glow. Just just my thought.

That said, I noticed that the logo of the actual pfSense firewall is not black and white, but looks like below;
![image](https://user-images.githubusercontent.com/16717388/80422480-8133e300-88de-11ea-8e99-0b9a83060568.png)
Perhaps LibreNMS should rather adopt this one instead of the black and white one. In that case the problem is solved. However this could still be an issue though for some other logos with mostly black shades in their design.

facuxt commented:
Update:
Improved table headings.
![image](https://user-images.githubusercontent.com/7550822/81176778-058c0300-8f7c-11ea-854b-dfe56219ad8e.png)

Changed the line color of Ping and Poller
![image](https://user-images.githubusercontent.com/7550822/81176824-19376980-8f7c-11ea-8f86-91f988cab434.png)

facuxt commented:
> I think it is because the eye/brain realizes the fact that the black is part of the gray and blue, making the need for the glow unnecessary in this case. Not sure about an all-black logo though, I think it will look strange without the glow. Just just my thought.
> 
> That said, I noticed that the logo of the actual pfSense firewall is not black and white, but looks like below;
> ![image](https://user-images.githubusercontent.com/16717388/80422480-8133e300-88de-11ea-8e99-0b9a83060568.png)
> Perhaps LibreNMS should rather adopt this one instead of the black and white one. In that case the problem is solved. However this could still be an issue though for some other logos with mostly black shades in their design.

I suggest you to try this modifications and see for yourself if you can see the logo.

facuxt commented:
@Jellyfrog @kkrumm1 @PipoCanaja any chances to see it merged in the following milestones? can you tell me if there is anything pending to do on my side?

thanks

SourceDoctor commented:
@facuxt therefore travis has to run successfull. You've to merge against master to fix the issues which travis has

facuxt commented:
> @facuxt therefore travis has to run successfull. You've to merge against master to fix the issues which travis has

Thanks, i did the merge from master and uploaded in this branch/pr

facuxt commented:
> @facuxt therefore travis has to run successfull. You've to merge against master to fix the issues which travis has

Any idea how to get this further down in the development line? 

Thanks

kkrumm1 commented:
We need people to test this out first. @librenms/reviewers

facuxt commented:
> We need people to test this out first. @librenms/reviewers

Thanks, i 

> We need people to test this out first. @librenms/reviewers

Great, i will follow updates on this PR, if you need me to fix something, ill be around.

facuxt commented:
@kkrumm1 what happened?

kkrumm1 commented:
Merged it. Look like you were ready.

facuxt commented:
> Merged it. Look like you were ready.

i mean the "revert" commit.. dont know what happened there.

TheGreatDoc commented:
@facuxt @kkrumm1 Are you sure this was ok to merge?

This is how looks now my install with dark theme:
![image](https://user-images.githubusercontent.com/32565115/82721173-2746f300-9cbb-11ea-8644-8bd2a97cd0b3.png)

kkrumm1 commented:
Yes, you need to clear your cache.

> On May 22, 2020, at 11:04 PM, TheGreatDoc <notifications@github.com> wrote:
> 
> ﻿
> @facuxt @kkrumm1 Are you sure this was ok to merge?
> 
> This is how looks now my install with dark theme:
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

-- 










CONFIDENTIALITY NOTICE: This email & attached documents may 
contain confidential information. All information is intended only for the 
use of the named recipient. If you are not the named recipient, you are not 
authorized to read, disclose, copy, distribute or take any action in 
reliance on the information and any action other than immediate delivery to 
the named recipient is strictly prohibited. If you have received this email 
in error, do not read the information and please immediately notify sender 
by telephone to arrange for a return of the original documents. If you are 
the named recipient you are not authorized to reveal any of this 
information to any other unauthorized person. If you did not receive all 
pages listed or if pages are not legible, please immediately notify sender 
by phone.

TheGreatDoc commented:
Seems still needs a little bit more work (or is just me?)
![image](https://user-images.githubusercontent.com/32565115/82721283-1e0a5600-9cbc-11ea-908a-48cab84013f3.png)

And Device->Ports section too:
![image](https://user-images.githubusercontent.com/32565115/82721405-18614000-9cbd-11ea-938a-e5beaee8c068.png)

I'll keep testing it a bit and will comment again if I see more places that needs to be updated

TheGreatDoc commented:
Device -> Alerts (When no alerts)
![image](https://user-images.githubusercontent.com/32565115/82721532-22377300-9cbe-11ea-96ee-d7b8fec0d358.png)

kkrumm1 commented:
@facuxt you think you can submit a PR to fix those pages @TheGreatDoc is talking about?

facuxt commented:
yes, ofcourse, i will try to replicate those issues and fix them.

El dom., 24 may. 2020 10:49, Kevin Krumm <notifications@github.com>
escribió:

> @facuxt <https://github.com/facuxt> you think you can submit a PR to fix
> those pages @TheGreatDoc <https://github.com/TheGreatDoc> is talking
> about?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/librenms/librenms/pull/11417#issuecomment-633233684>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZTOZU254QIVPLPOSXT6F3RTEQUZANCNFSM4MHH6CGQ>
> .
>

murrant commented:
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-64-release-changelog-may-2020/12290/1

BloodyIron commented:
I've switched the Dark Theme on and been loving it for a long while now! However there is one area I really would like to get adjusted, rrd related. Specifically I'm talking about Apps (OS Updates/Proxmox/etc).

When I go into the "OS Updates" section of Apps, only the single day graphs are shown, not multiple graphs per device. But this does look good with the dark theme. So if we could get more graphs on this page for history, that would be great, in similar vein to how most other areas of libreNMS do it.

When I go into the Overview section of Apps, the graphs that I see there have white background with green graph, unlike the black background in the OS Updates section itself.  These graphs in the Overview section show up for OS Updates and Proxmox (I don't have other App stuff setup yet). So this really counters the point of a dark theme for this section.

If we could get this addressed at some point that'd be great. :)
